### PR TITLE
Memory leak & some cleanup

### PR DESCRIPTION
**Pull request by bkueng**
@julianoes can you test if this works on Snapdragon?

### CODE REVIEW

**Comment by julianoes** [comment]
Thanks, will do.
I think you'll need to rebase on latest master. Travis didn't like 0/0 :).

File: src/platforms/posix/px4_layer/px4_posix_tasks.cpp [hunk 1]
```diff
@@ -229,6 +234,13 @@ px4_task_t px4_task_spawn_cmd(const char *name, int scheduler, int priority, int
 		}
 	}
 
+	if (i >= PX4_MAX_TASKS) {
```
Comment (julianoes) [inline src/platforms/posix/px4_layer/px4_posix_tasks.cpp]: Shouldn't this check happen before a task is started?

**Comment by julianoes** [comment]
Tested on Snappy, still works, nothing broken.

File: src/platforms/posix/px4_layer/px4_posix_tasks.cpp (hunk 1 above)
Comment (bkueng) [inline src/platforms/posix/px4_layer/px4_posix_tasks.cpp]: do you mean the thread? The thread is started afterwards.

**Comment by bkueng** [comment]
Thanks!

File: src/platforms/posix/px4_layer/px4_posix_tasks.cpp (hunk 1 above)
Comment (julianoes) [inline src/platforms/posix/px4_layer/px4_posix_tasks.cpp]: Aha, ok. I guess you got it covered ;).

**Pull request merged**